hrjobcontract/tests/phpunit/api/v3/HRJobContractRevisionTest.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+use Civi\Test\HeadlessInterface;
+use Civi\Test\TransactionalInterface;
+
+use CRM_HRCore_Test_Fabricator_Contact as ContactFabricator;
+use CRM_Hrjobcontract_Test_Fabricator_HRJobContract as HRJobContractFabricator;
+
+/**
+ * Class api_v3_HRJobContractRevisionTest
+ *
+ * @group headless
+ */
+class api_v3_HRJobContractRevisionTest extends PHPUnit_Framework_TestCase implements
+  HeadlessInterface,
+  TransactionalInterface {
+
+  public function setUpHeadless() {
+    return \Civi\Test::headless()
+      ->install('uk.co.compucorp.civicrm.hrcore')
+      ->installMe(__DIR__)
+      ->apply();
+  }
+
+  public function testItAcceptsBothIntegerAndStringForChangeReasonValue() {
+    $contact = ContactFabricator::fabricate();
+    $contract = HRJobContractFabricator::fabricate(['contact_id' => $contact['id']], ['period_start_date' => '2015-01-01']);
+    $reason = $this->getReasonChangeOption();
+
+    $revision = $this->createRevision($contract['id'], $reason);
+    $this->assertEquals($revision['is_error'], 0);
+
+    $reason = $this->updateReasonChangeOptionValue($reason['id'], 'Lorem Ipsum');
+
+    $revision = $this->createRevision($contract['id'], $reason);
+    $this->assertEquals($revision['is_error'], 0);
+  }
+
+  /**
+   * Creates a revision via the api
+   *
+   * @param  [int] $contract_id
+   * @param  [Array] $reason    The reason change option
+   * @return [Array]
+   */
+  private function createRevision($contract_id, $reason) {
+    return civicrm_api3('HRJobContractRevision', 'create', array(
```

Comment:
@omarabuhussein @davialexandre the reason I didn't use a fabricator here is that this is explicitly about using the api to create a revision, while a fabricator hide the implementation details from you, so you don't know if it's using the api, the BAO, raw sql, etc.

I'm not sure this is the optimal approach, let me know what you think